resources/migrations/custom_sql/fill_cache_config.h2.sql

```diff
@@ -0,0 +1,44 @@
+WITH root AS (
+  select 'root' AS model,
+         0      AS model_id,
+         'ttl'  AS strategy,
+         json_object(
+           'multiplier' VALUE coalesce((select ("VALUE"::varchar(15))::int from setting where "KEY" = 'query-caching-ttl-ratio'), 10),
+           'min_duration_ms' VALUE coalesce((select ("VALUE"::varchar(15))::int from setting where "KEY" = 'query-caching-min-ttl'), 60000)
+         ) AS config
+), database AS (
+  select 'database' AS model,
+         id         AS model_id,
+         'duration' AS strategy,
+         json_object('duration' VALUE cache_ttl, 'unit' VALUE 'hours') AS config
+    from metabase_database
+   where cache_ttl is not null
+), dashboard AS (
+  select 'dashboard' AS model,
+         id          AS model_id,
+         'duration'  AS strategy,
+         json_object('duration' VALUE cache_ttl, 'unit' VALUE 'hours') AS config
+    from report_dashboard
+   where cache_ttl is not null
+), card AS (
+  select 'question' AS model,
+         id         AS model_id,
+         'duration' AS strategy,
+         json_object('duration' VALUE cache_ttl, 'unit' VALUE 'hours') AS config
+    from report_card
+   where cache_ttl is not null
+), rows1 AS (
+  SELECT * FROM root UNION ALL
+  SELECT * FROM database UNION ALL
+  SELECT * FROM dashboard UNION ALL
+  SELECT * FROM card
+), rows AS (
+  SELECT * FROM rows1
+   WHERE (SELECT true FROM setting WHERE "KEY" = 'enable-query-caching' AND "VALUE" = 'true')
```

Comment:
minor thing: I haven't looked at the query plan, but is H2 smart enough to short circuit everything if this is false? Seems trivial to push down to individual queries if not.

Reply:
As discussed offline I feel a bit weird about us "dropping" any latent config when the customer upgrades to this version, and worry a bit that they'd built up a complex configuration and only temporarily disabled it. Since we're phasing out the UI, and not yet deleting the source data though I guess it's fine - if any customers escalate that they want to recover their config Success can just share this query without the filter for them to run manually.

Reply:
> is H2 smart enough to short circuit everything if this is false

I think it's more important to keep query simpler/more understandable, there's not going to be millions of rows anyway.